### PR TITLE
feat(trace): exception event

### PR DESCRIPTION
**Pull request by RogerHYang**
resolves: #1056

### CODE REVIEW

File: src/phoenix/trace/semantic_conventions.py [hunk 1]
```diff
@@ -38,6 +38,8 @@ class DeploymentAttributes(AttributeGroup):
     }
 
 
+EXCEPTION_MESSAGE = "exception.message"
```
Comment (mikeldking) [inline src/phoenix/trace/semantic_conventions.py:41]: If we are following OTEL, this should be added to the events as an exception event: https://github.com/Arize-ai/phoenix/blob/e5f716e1188b5289cc28d2c462952b8a4dc780e1/src/phoenix/trace/schemas.py#L65

**Review by mikeldking — CHANGES_REQUESTED** [review]
I think this diverges from the `events` exception type I believe. LMK

File: src/phoenix/experimental/callbacks/langchain_tracer.py [hunk 2]
```diff
@@ -81,17 +81,28 @@ def _convert_run_to_spans(
...
+            status_code = SpanStatusCode.OK
+        else:
+            status_code = SpanStatusCode.ERROR
+            error_event = next(filter(lambda event: event["name"] == "error", run["events"]))
+            events.append(
+                SpanException(
+                    message=error,
+                    timestamp=error_event["time"],
+                )
+            )
```
Comment (mikeldking) [inline src/phoenix/experimental/callbacks/langchain_tracer.py:95]: doesn't this only record one event but in reality we might want to record all the exceptions? Also please feel free to extend the event with callstack etc. as noted in the spec.

**Review by mikeldking — APPROVED** [review]

File: src/phoenix/experimental/callbacks/langchain_tracer.py (hunk 2 above)
Comment (RogerHYang) [reply src/phoenix/experimental/callbacks/langchain_tracer.py]: you're right i shouldn't assume there's only one exception. i updated the code. on the other hand, we don't have access to the call stack with the current approach which is to simply transcribe the final langsmith result. To get to the actual call stack we would need to more "high touch" version, i.e. making and keeping our spans in a stack and so on.

Comment (RogerHYang) [reply src/phoenix/experimental/callbacks/langchain_tracer.py]: actually, i think the langsmith structure only keep a single copy of the error message, so if there are more than one error event, i don't see that they would have separate messages

Comment (RogerHYang) [reply src/phoenix/experimental/callbacks/langchain_tracer.py]: added a comment to clarify why only one error event is kept

**Pull request merged**